### PR TITLE
Save HR data on failfast

### PR DESCRIPTION
**Pull request by tcare**

### CODE REVIEW

File: lib/Common/Core/Assertions.h [hunk 1]
```diff
@@ -77,10 +77,12 @@ extern int IsInAssert;
 
 #ifdef DBG
 #define AssertOrFailFast(x)                 Assert(x)
+#define AssertOrFailFastHR(x, hr)           Assert(x)
 #define AssertOrFailFastMsg(x, msg)         AssertMsg(x, msg)
```
Comment (akroshg) [inline lib/Common/Core/Assertions.h]: do you need the HR part for this one as well?

Comment (tcare) [reply lib/Common/Core/Assertions.h]: I think it's not needed in this case, it's more that the rethrow by failfast loses the context info.

Comment (dilijev) [reply lib/Common/Core/Assertions.h]: Might as well add it; we use AssertOrFailFastMsg a lot, and in addition to the change below, I see this change enabling folks to write FailFast-with-HR in the future.

**Comment by tcare** [comment]
Merging manually due to ChakraHub issue.

**Pull request closed without merging**